tmt/utils.py

```diff
@@ -3619,32 +3619,86 @@ def _create_directory() -> tuple[str, Optional[Exception]]:
 
 
 def create_file(
+        *,
         path: Path,
         content: str,
         name: str,
         dry: bool = False,
         force: bool = False,
         mode: int = 0o664,
-        quiet: bool = False) -> None:
-    """ Create a new file, handle errors """
-    say = log.debug if quiet else echo
-    action = 'would be created' if dry else 'created'
-    if path.exists():
-        if force:
-            action = 'would be overwritten' if dry else 'overwritten'
-        else:
-            raise FileError(f"File '{path}' already exists.")
+        quiet: bool = False,
+        logger: tmt.log.Logger) -> None:
+    """
+    Create a new file.
+
+    Before creating the file, function checks whether it exists
+    already - the existing file is **not** removed and re-created,
+    unless ``force`` is set.
+
+    The outcome of the operation will be logged in a debug log, but
+    may also be sent to console with ``quiet=False``.
+
+    :param path: a path to be created.
+    :param content: content to save into the file
+    :param name: a "label" of the path, used for logging.
+    :param dry: if set, the file would not be created or overwritten. Still,
+        the existence check will happen.
+    :param force: if set, the file would be overwritten if it already exists.
+    :param mode: permissions to set for the file.
+    :param quiet: if set, an outcome of the operation would be logged
+        to console.
```

Comment:
Maybe I got it wrong, but shouldn't it be the opposite? If NOT set (`quiet=False`), an outcome of the operation would be logged to console.

Reply:
Unfortunately, it's correct :/ And yes, it's puzzling, but `quiet` relates to logging only. Both `create_directory()` and `create_file()` combine logging (that goes to stderr) and reporting to the user (stdout), and my changes try to preserve the behavior when logging might be suppressed.

I can update docstring to improve this, let's say we merged this PR & I fix both functions?

Reply:
If I understand it correctly, the operation is logged to the debug log (using the Logger) regardless. However, it may also be sent to the console if `quiet=False`. If `quiet=True`, the operation should not be printed to the console. A few lines above, there is this docstring:

> The outcome of the operation will be logged in a debug log, but may also be sent to console with ``quiet=False``.

That makes sense. However, the `quiet` parameter seems to suggest that the operation is printed to the console only if the parameter is set to True, which is the opposite.

> if set, an outcome of the operation would be logged to console.

That's why I'm confused. But if you say it's correct, I would appreciate if the docstring could be improved a bit.

Reply:
Hopefully fixed in 7d878979